src/applications/facility-locator/utils/SearchAreaControl.js

```diff
@@ -0,0 +1,24 @@
+export default class SearchAreaControl {
+  constructor(isMobile) {
+    this.isMobile = isMobile;
+  }
+
+  onAdd(map) {
+    this.map = map;
+    this.container = document.createElement('div');
+    this.container.id = 'search-area-control-container';
+    this.container.className = this.isMobile
+      ? 'mapboxgl-ctrl-bottom-center'
+      : 'mapboxgl-ctrl-top-center';
+    this.container.style.display = 'none';
+    this.button = document.createElement('button');
+    this.button.id = 'search-area-control';
+    this.button.textContent = 'Search this area of map';
+    this.container.appendChild(this.button);
+    return this.container;
+  }
+  onRemove() {
+    this.container.parentNode.removeChild(this.container);
+    this.map = undefined;
+  }
+}
```

Comment:
This works, but why not use a React component?

Reply:
Because the map library is not react is javascript, same story with the markers - https://codepen.io/sergei-zelinsky/pen/gGOZjE

Reply:
https://docs.mapbox.com/help/tutorials/custom-markers-gl-js/

Reply:
For the Markers, we are using [react-leaflet](https://react-leaflet.js.org/).

For the Search button, we might need something like [react-mapbox-gl](https://www.npmjs.com/package/react-mapbox-gl). But we can iterate on this. The code you have written here works fine for now.